RoT/Protocol/RFCs/0001-Transport_Agnosticism.md

```diff
@@ -0,0 +1,149 @@
+* Name: Transport_Agnosticism
+* Date: 2021-04-02
+* Pull Request: [#16](https://github.com/opencomputeproject/Security/pull/16)
+
+# Objective
+
+Cerberus currently mandates that its messages be exchanged over an MCTP bus,
+using a Microsoft-specific PCI vendor ID. This is a limitation for depoyments
+that might want to speak Cerberus over a completely different bus, such as
+SPI, I3C, exotic busses like NVME, or over plain old TCP or UDP for the
+purposes of conformance testing.
+
+This RFC describes a "transport agnostic" model for Cerberus with the following
+goals:
+- The existing MCTP binding of the protocol works with minimal (or no) changes.
+- Cerberus can be used over an arbitrary transport layer without directly
+  implementing special support for it in a generic Cerberus library.
+- References to MCTP in the Cerberus Challenge Protocol specification will be
+  moved to an appendix, describing it as a possible option for transporting
+  Cerberus.
+
+# Proposal
+
+"Transport agnostic" Cerberus specifies an *admissible transport* as any protocol
+or bus that satisfies a set of properties. These properties reflect the
+properties of MCTP already used by Cerberus.
+
+1. An admissible transport is a mechanism for sending a *message* (a dynamically-\
+   sized byffer of bytes) from one (not necesarily addressable) endpoint to
+   another. In other words, the admissible transport is responsible for removing
+   frames from packets and assembling them in sequence.
+
+2. Each endpoint has a known *maximum message length*, measured in bytes. The
+   transport is responsible for negotiating this parameter for each device a
+   Cerberus device intends to speak to. This parameter must be made available
+   to the Cerberus protocol but Cerberus itself does not negotiate it. This
```

Comment:
We could. I'm not sure there's an immediate thing this restriction buys us (it certainly doesn't benefit the client in any way I can see...).

Reply:
Maybe things change when you move to non-I2C/MTCP busses, but having a hard cap on the maximum message size is useful, especially when talking about hierarchical attestation.  The PA-RoT knows if it allocates a message buffer of 4096 bytes, it will be able to fully communicate with any device.  It also helps buffer sizing in the BMC to support routing messages to/from different devices within the system (e.g. PA-RoT <-> AC-RoT).